### ISSUE TITLE
fragment-matcher generating incorrect possibleTypes signature

### ISSUE THREAD

**Issue by jagregory** [dotansimha/graphql-code-generator#1028]
**Describe the bug**

The new fragment-matcher plugin is generating a results object which is incompatible with the interface signature it also generates.

The output file contains types for all the types in the schema and ([correctly according to the spec](https://github.com/facebook/graphql/blob/master/spec/Section%204%20--%20Introspection.md#schema-introspection)) has `possibleTypes` set to `null` for anything which isn't a Union/Interface; however, the generated `IntrospectionResultData` interface doesn't allow `null` so it doesn't compile.

**To Reproduce**

**schema**
```graphql
type Query {
  hello: String
}
```

**config**
```yaml
schema: ./graphql/schema.graphql
documents: ./src/**/*.graphql
generates:
  src/fragment-matcher.ts:
    plugins:
    - fragment-matcher
```

**fragments file**
```ts
export interface IntrospectionResultData {
  __schema: {
    types: {
      kind: string;
      name: string;
      possibleTypes: {
        name: string;
      }[];
    }[];
  };
}

const result: IntrospectionResultData = {
  __schema: {
    types: [
      {
        kind: "OBJECT",
        name: "Query",
        possibleTypes: null
      },
      {
        kind: "SCALAR",
        name: "String",
        possibleTypes: null
      },
      {
        kind: "OBJECT",
        name: "__Schema",
        possibleTypes: null
      },
      {
        kind: "OBJECT",
        name: "__Type",
        possibleTypes: null
      },
      {
        kind: "ENUM",
        name: "__TypeKind",
        possibleTypes: null
      },
      {
        kind: "SCALAR",
        name: "Boolean",
        possibleTypes: null
      },
      {
        kind: "OBJECT",
        name: "__Field",
        possibleTypes: null
      },
      {
        kind: "OBJECT",
        name: "__InputValue",
        possibleTypes: null
      },
      {
        kind: "OBJECT",
        name: "__EnumValue",
        possibleTypes: null
      },
      {
        kind: "OBJECT",
        name: "__Directive",
        possibleTypes: null
      },
      {
        kind: "ENUM",
        name: "__DirectiveLocation",
        possibleTypes: null
      }
    ]
  }
};

export default result;
```

**Expected behavior**
Either non-Union/Interface types should be excluded from the result object, or the `possibleTypes` signature should be changed to allow `null`.

**Environment:**
 - Codegen: 0.15.0

**Comment by jagregory** [comment]
Actually, a little more info: Apollo has `IntrospectionResultData` defined as:

```ts
export declare type IntrospectionResultData = {
    __schema: {
        types: {
            kind: string;
            name: string;
            possibleTypes: {
                name: string;
            }[];
        }[];
    };
};
```

Which means that you can't change the type definition generated by fragment-matcher, otherwise you can't pass the generated result to ApolloClient.

So it looks like the only real option is filtering non-union and non-interfaces out?

**Comment by dotansimha** [comment]
Fixed in 0.15.1
